devices/common_patches/l4_adc_common.yaml

```
@@ -0,0 +1,166 @@
+            bitOffset: 17
+            bitWidth: 1
+          ADRDY_SLV:
+            description: Slave ADC ready
+            bitOffset: 16
+            bitWidth: 1
+
+          JQOVF_MST:
+            description: Injected Context Queue Overflow flag of the master ADC
+            bitOffset: 10
+            bitWidth: 1
+          AWD3_MST:
+            description: Analog watchdog 3 flag of the master ADC
+            bitOffset: 9
+            bitWidth: 1
+          AWD2_MST:
+            description: Analog watchdog 2 flag of the master ADC
+            bitOffset: 8
+            bitWidth: 1
+          AWD1_MST:
+            description: Analog watchdog 1 flag of the master ADC
+            bitOffset: 7
+            bitWidth: 1
+          JEOS_MST:
+            description: End of injected sequence flag of the master ADC
+            bitOffset: 6
+            bitWidth: 1
+          JEOC_MST:
+            description: End of injected conversion flag of the master ADC
+            bitOffset: 5
+            bitWidth: 1
+          OVR_MST:
+            description: Overrun flag of the master ADC
+            bitOffset: 4
+            bitWidth: 1
+          EOS_MST:
+            description: End of regular sequence flag of the master ADC
+            bitOffset: 3
+            bitWidth: 1
+          EOC_MST:
+            description: End of regular conversion flag of the master ADC
+            bitOffset: 2
+            bitWidth: 1
+          EOSMP_MST:
+            description: End of Sampling phase flag of the master ADC
+            bitOffset: 1
+            bitWidth: 1
+          ADRDY_MST:
+            description: master ADC ready
+            bitOffset: 0
+            bitWidth: 1
+
+      CCR:
+        description: ADC common control register
+        addressOffset: 0x08
+        access: read-write
+        resetValue: 0x00000000
+        fields:
+          CH18SEL:
+            description: CH18 selection (Vbat)
+            bitOffset: 24
+            bitWidth: 1
+          CH17SEL:
+            description: CH17 selection (temperature)
+            bitOffset: 23
+            bitWidth: 1
+          VREFEN:
+            description: Vrefint enable
+            bitOffset: 22
+            bitWidth: 1
+          PRESC:
+            description: ADC prescaler
+            bitOffset: 18
+            bitWidth: 4
+          CKMODE:
+            description: ADC clock mode
+            bitOffset: 16
+            bitWidth: 2
+          MDMA:
+            description: Direct memory access mode for dual ADC mode
+            bitOffset: 14
+            bitWidth: 2
+          DMACFG:
+            description: DMA configuration (for dual ADC mode)
+            bitOffset: 13
+            bitWidth: 1
+          DELAY:
+            description: Delay between 2 sampling phases
+            bitOffset: 8
+            bitWidth: 4
+          DUAL:
+            description: Dual ADC mode selection
+            bitOffset: 0
+            bitWidth: 4
```

I think this should be 5 bits?
```suggestion
            bitWidth: 5
```